tests/Keywords/CucumberKeywordsTest.php

```diff
@@ -39,4 +40,53 @@ protected function getSteps($keywords, $text, &$line, $keywordType)
 
         return $steps;
     }
+
+    public function testYamlSourceFileIsAttachedToException(): void
+    {
+        $tempFile = tempnam(sys_get_temp_dir(), 'test');
+
+        try {
+            file_put_contents($tempFile, "invalid:\n\tinvalid:yaml");
+
+            $this->expectExceptionObject(new ParseException(
+                'YAML file cannot contain tabs as indentation',
+                2,
+                "\tinvalid:yaml",
+                $tempFile,
+            ));
+
+            new CucumberKeywords($tempFile);
+        } finally {
+            @unlink($tempFile);
+        }
+    }
+
+    public function testYamlRootMustBeAnArray(): void
+    {
+        $this->expectException(ParseException::class);
+        $this->expectExceptionMessage('Root element must be an array, but string found.');
+
+        new CucumberKeywords("a\nstring");
+    }
+
+    public function testYamlFileMustBeReadable(): void
+    {
+        $tempFile = tempnam(sys_get_temp_dir(), 'test');
+
+        try {
+            file_put_contents($tempFile, "aaa:\n  bbb: cccc");
+            if (PHP_OS_FAMILY === 'Windows') {
+                exec('icacls ' . escapeshellarg($tempFile) . ' /deny Everyone:(R)');
```

Comment:
I can confirm this works on Windows. I'm guessing the unix variation below should also work.

Reply:
True, I just checked - the functions (`is_file`, `is_readable`) we're using should support stream wrappers, so that should also work. That would also get rid of necessitating a random file in the system temp dir. 🤔

Reply:
I'll keep it as is, since it works. If we see any issues, we can go through the additional effort of switching to VFS.